sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2201,9 +2202,19 @@ object GpuOverrides {
     exec[BatchScanExec](
       "The backend for most file input",
       ExecChecks(
-        (TypeSig.commonCudfTypes + TypeSig.STRUCT + TypeSig.MAP + TypeSig.ARRAY).nested(),
+        (TypeSig.commonCudfTypes + TypeSig.STRUCT + TypeSig.MAP + TypeSig.ARRAY +
+            TypeSig.DECIMAL).nested(),
         TypeSig.all),
       (p, conf, parent, r) => new SparkPlanMeta[BatchScanExec](p, conf, parent, r) {
+        override def tagPlanForGpu(): Unit = {
```

Comment:
This should not be needed because the Scan should tag itself as not supporting decimal.

Reply:
Yes, it is unnecessary. I've removed it.

Reply:
The problem with removing this code is that the scans **are not** checking for decimal types correctly.  For example, I just tried out this patch on an ORC file with decimals which should not run with the plugin because the ORC reader doesn't support decimals yet.  Note how it says it is going to run the ORC load on the GPU:
```
scala> val df = spark.read.orc("/home/jlowe/tmp/orc-decimals.orc")
df: org.apache.spark.sql.DataFrame = [c_0: decimal(18,0), c_1: decimal(7,3) ... 4 more fields]

scala> df.explain
20/12/18 16:17:04 WARN GpuOverrides: 
*Exec <FileSourceScanExec> could run on GPU

== Physical Plan ==
GpuColumnarToRow false
+- GpuFileGpuScan orc [c_0#54,c_1#55,c_2#56,c_3#57,c_4#58L,c_5#59] Batched: true, DataFilters: [], Format: ORC, Location: InMemoryFileIndex[file:/home/jlowe/tmp/orc-decimals.orc], PartitionFilters: [], PushedFilters: [], ReadSchema: struct<c_0:decimal(18,0),c_1:decimal(7,3),c_2:decimal(10,10),c_3:decimal(15,12),c_4:bigint,c_5:fl...
```

I agree with @revans2 that this change really shouldn't be here, as the scans should be responsible for properly tagging whether they can load the types or not.  However in the past the scans all supported the same types and were relying on the parent exec type checking to check the types.  Now that the scans don't all support the same types, they need to handle the type check explicitly in the separate `tagSupport` methods of `Scan[CSVScan]`, `Scan[OrcScan]`, and `Scan[ParquetScan]`.

Please add the appropriate type checking code in those places along with some fallback tests on CSV and ORC that verify we're not trying to load these on the GPU when decimals are involved.  If we get this wrong it won't crash, and the data will be silently corrupted on load instead.

Reply:
Hi @jlowe, I've changed type checking codes of CSV and ORC reader to disable decimal reading. And corresponding test case is also provided.